galsim/chromatic.py

```diff
@@ -421,6 +421,97 @@ def drawImage(self, bandpass, image=None, integrator='trapezoidal', **kwargs):
+        _remove_setup_kwargs(kwargs)
+
+        # determine combined self.wave_list and bandpass.wave_list
+        wave_list, _ , _ = galsim.utilities.combine_wave_list(self, bandpass)
+
+        if self.separable:
+            if len(wave_list) > 0:
+                multiplier = np.trapz(self.SED(wave_list) * bandpass(wave_list), wave_list)
+            else:
+                multiplier = galsim.integ.int1d(lambda w: self.SED(w) * bandpass(w),
+                                                bandpass.blue_limit, bandpass.red_limit)
+            prof0 *= multiplier/self.SED(bandpass.effective_wavelength)
+            image = prof0.drawKImage(image=image, **kwargs)
+            return image
+
+        # Decide on integrator.  If the user passed one of the integrators from galsim.integ, that's
+        # fine.  Otherwise we decide based on the adopted integration rule and the presence/absence
+        # of `wave_list`.
+        if isinstance(integrator, str):
+            if integrator == 'trapezoidal':
+                rule = np.trapz
+            elif integrator == 'midpoint':
+                rule = galsim.integ.midpt
+            else:
+                raise TypeError("Unrecognized integration rule: %s"%integrator)
+            if len(wave_list) > 0:
+                integrator = galsim.integ.SampleIntegrator(rule)
+            else:
+                integrator = galsim.integ.ContinuousIntegrator(rule)
+        if not isinstance(integrator, galsim.integ.SampleIntegrator) and \
+                not isinstance(integrator, galsim.integ.ContinuousIntegrator):
+            raise TypeError("Invalid type passed in for integrator!")
+
+        # merge self.wave_list into bandpass.wave_list if using a sampling integrator
+        if isinstance(integrator, galsim.integ.SampleIntegrator):
+            bandpass = galsim.Bandpass(galsim.LookupTable(wave_list, bandpass(wave_list),
+                                                          interpolant='linear'),
+                                       wave_type='nm')
+
+        add_to_image = kwargs.pop('add_to_image', False)
+        image_int = integrator(self.evaluateAtWavelength, bandpass, image, kwargs, doK=True)
+
+        # For performance profiling, store the number of evaluations used for the last integration
+        # performed.  Note that this might not be very useful for ChromaticSum instances, which are
+        # drawn one profile at a time, and hence _last_n_eval will only represent the final
+        # component drawn.
+        self._last_n_eval = integrator.last_n_eval
+
+        # Apply integral to the initial image appropriately.
+        # Note: Don't do image = integral and return that for add_to_image==False.
+        #       Remember that python doesn't actually do assignments, so this won't update the
+        #       original image if the user provided one.  The following procedure does work.
+        if add_to_image:
+            image += image_int
```

Comment:
Oh right.  I meant it would be surprising to actually do a recenter on a not-to-be-zeroed image.  (If it already exists, then we shouldn't be messing with its wcs, right?)  If the image is already centered, then `recenter=True` should always be fine, regardless of the state of `add_to_image`.

So maybe the condition I want to raise on is actually:
```python
if recenter and image.center() != galsim.PositionI(0,0) and add_to_image:
    raise ValueError("Cannot recenter a non-centered image when add_to_image=True")
```